### PR TITLE
DOC: Deprecate null_counts parameter of DataFrame.info

### LINKED ISSUE #36805: DOC: unclear kwarg null_counts
#### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.info.html?highlight=info#pandas.DataFrame.info

#### Documentation problem

Keyword ``null_counts`` from its name may be considered as a need to count null items.
However, the documentation explains that is quite the opposite - count **non-null** items.
I suggest to rename it with ``show_counts`` and maintain the old behavior.

#### Suggested fix for documentation

Replace ``null_counts`` with ``show_counts``.
It will make it more clear what this kwarg means without even the need to read docs.

#### API

It the suggested rename is implemented, then need to start deprecation of kwarg ``null_counts``.

### PR DESCRIPTION
**Pull request by hongshaoyang**
- [x] closes #36805
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### Background. 

> Might be worth it actually - I _think_ people are only likely to use `.info` in an interactive session, rather than as part of production code, so perhaps it is worth going through the deprecation process to make this argument clearer

* xref https://github.com/pandas-dev/pandas/issues/36805#issuecomment-702754290
* Deprecate `null_counts` parameter of DataFrame.info in favor of `show_counts` and add warning.
* Add whatsnew and add tests to test for FutureWarning. Move usages of null_counts in tests to show_counts.

Doc preview: 

Screenshot 2020-11-22 at 6 19 05 PM

### CODE REVIEW

**Review by arw2019 — COMMENTED** [review]
Thanks @hongshaoyang  for the PR!

We'll want to catch this warning in any tests that hit this. 

We'll also need a whatsnew

**Comment by hongshaoyang** [comment]
Thanks @arw2019 , I added tests and a whatsnew entry.

**Comment by jreback** [comment]
looks fine, @jorisvandenbossche ok here?

**Comment by jreback** [comment]
just merged some changes that will affect this PR, can you merge master and update: https://github.com/pandas-dev/pandas/pull/37868

File: pandas/tests/io/formats/test_format.py [hunk 1]
```diff
@@ -194,6 +194,11 @@ def check(null_counts, result):
             check(True, False)
             check(False, False)
 
+        with tm.assert_produces_warning(FutureWarning):  # GH37999
```
Comment (ivanovmg) [inline pandas/tests/io/formats/test_format.py:197]: ``tm.assert_produces_warning`` now has an optional ``match`` kwarg to match the warning message.
Could use it here to ensure that the proper warning message is issues.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -2639,8 +2642,16 @@ def info(
...
         memory_usage: Optional[Union[bool, str]] = None,
+        show_counts: Optional[bool] = None,
         null_counts: Optional[bool] = None,
     ) -> None:
+        if null_counts is not None and show_counts is None:
+            warnings.warn(
+                "null_counts is deprecated. Use show_counts instead",
+                FutureWarning,
+                stacklevel=2,
+            )
```
Comment (ivanovmg) [inline pandas/core/frame.py:2653]: The user will not get this warning when setting both arguments at the same time.
For example, ``df.info(null_counts = True, show_counts=False)``.
This type of invocation is incorrect and should raise an error.

File: pandas/tests/io/formats/test_format.py (hunk 1 above)
Comment (hongshaoyang) [reply pandas/tests/io/formats/test_format.py]: Added

File: pandas/core/frame.py (hunk 2 above)
Comment (hongshaoyang) [reply pandas/core/frame.py]: Added

File: pandas/core/frame.py [hunk 3]
```diff
@@ -2639,8 +2642,18 @@ def info(
         buf: Optional[IO[str]] = None,
         max_cols: Optional[int] = None,
         memory_usage: Optional[Union[bool, str]] = None,
+        show_counts: Optional[bool] = None,
         null_counts: Optional[bool] = None,
     ) -> None:
+        if null_counts is not None:
+            if show_counts is not None:
+                raise ValueError("null_counts used with show_counts")
```
Comment (ivanovmg) [inline pandas/core/frame.py:2650]: I suggest adding ``Use show_counts``.
Otherwise looks good to me.

**Review by ivanovmg — APPROVED** [review]
One minor comment. Otherwise good.

File: doc/source/whatsnew/v1.2.0.rst [hunk 4]
```diff
@@ -480,6 +480,7 @@ Deprecations
 - The ``how`` keyword in :meth:`PeriodIndex.astype` is deprecated and will be removed in a future version, use ``index.to_timestamp(how=how)`` instead (:issue:`37982`)
 - Deprecated :meth:`Index.asi8` for :class:`Index` subclasses other than :class:`DatetimeIndex`, :class:`TimedeltaIndex`, and :class:`PeriodIndex` (:issue:`37877`)
 - The ``inplace`` parameter of :meth:`Categorical.remove_unused_categories` is deprecated and will be removed in a future version (:issue:`37643`)
+- The ``null_counts`` parameter of :meth:`DataFrame.info` is deprecated and will be removed in a future version (:issue:`37999`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.2.0.rst:483]: Can you indicate here that the keyword is renamed to `show_counts` ?

Comment (hongshaoyang) [reply doc/source/whatsnew/v1.2.0.rst]: Added

Comment (hongshaoyang) [reply doc/source/whatsnew/v1.2.0.rst]: There's an unrelated test on Travis that fails intermittently


```
FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_area_lim
```

**Comment by jreback** [comment]
thanks @hongshaoyang very nice

**Comment by jbrockmendel** [comment]
@hongshaoyang want to make a PR enforcing this deprecation?

**Pull request merged**